fix(Logo-grid): changed the card heading style to expressive

BrunnoM7 opened the pull request:
### Related Ticket(s)

[Logo grid] without expressive theme, card link type needs update #6211

### Description

{{Add a human-readable description / detail summary of what the PR is changing and any details around how and why}}

{{If applicable, include a screenshot indicating an example or examples of what the PR is changing in the application}}

### Changelog

**New**

- {{new thing}}

**Changed**

- {{changed thing}}

**Removed**

- {{removed thing}}

<!-- React and Web Component deploy previews are enabled by default. -->
<!-- To enable additional available deploy previews, apply the following -->
<!-- labels for the corresponding package: -->
<!-- *** "package: services": Services -->
<!-- *** "package: utilities": Utilities -->
<!-- *** "package: styles": Carbon Expressive -->
<!-- *** "RTL": React / Web Components (RTL) -->
<!-- *** "feature flag": React / Web Components (experimental) -->

jeffchew commented:
Thanks @BrunnoM7 ! Be sure to add the `Needs design approval` and `eyes needed` labels too.

jeffchew commented on packages/styles/scss/components/logo-grid/_logo-grid.scss:
```diff
@@ -139,7 +139,7 @@
     .#{$prefix}--content-block__cta
     .#{$prefix}--card__heading {
-    @include carbon--type-style('productive-heading-01');
+    @include carbon--type-style('expressive-heading-02');
```
@BrunnoM7 if this is the React version of LogoGrid, it might be an issue with the way the card link is used. @shixiedesign identified that the card link's heading in general was incorrect, and I noticed that `CardLink` was passing in `copy` instead of `heading`. See PR here: https://github.com/carbon-design-system/carbon-for-ibm-dotcom/pull/7056.

shixiedesign commented:
Web components looked fine but unable to check React because there's no preview link 😢 @BrunnoM7. Lemme know when that's ready and I can take another look.

BrunnoM7 commented:
Hey @shixiedesign, I updated the `styles` package to work with the PR Jeff mentioned (#7056). I changed the parameter of the CTA text to passed as copy, the style won't look right without this other PR since I also removed the rule that was overriding the `cardLink` style in React. I also added the knobs for the CTA to both React and Web Components.

jeffchew commented on packages/react/src/components/LogoGrid/LogoGrid.js:
```diff
@@ -31,7 +31,7 @@ const LogoGrid = ({ heading, logosGroup, ctaCopy, ctaHref, hideBorder }) => {
       style: 'card',
       type: 'local',
-      heading: ctaCopy,
+      copy: ctaCopy,
```
Card CTA should be passed as `heading` now.

jeffchew commented on .github/workflows/percy-update-base.yml:
```diff
@@ -5,6 +5,7 @@ on:
     branches:
       - master
       - release/*
+      - feat/expressive-sunset
```
This should be removed now this is targeting `master`.

emyarod reviewed (approved):
looks good to me! pending visual review

shixiedesign reviewed (approved):
Looks good! Thank you!
